adminSiteServer/apiRouter.ts

```diff
@@ -948,6 +948,36 @@ apiRouter.get("/variables.json", async (req) => {
     return { variables: rows, numTotalRows: numTotalRows }
 })
 
+apiRouter.get("/variables.usages.json", async (req) => {
+    // This is a bit of a workaround - what we would like is to select
+    // all variables and do a group by on the joined charts table with a count(*)
+    // but since the variables are stored inside the json config and mysql 5.7's limited
+    // support for json and arrays we can'd do it that way - so instead we select the arrays
+    // and create a map that we use to "transpose" the table, then serialize this
+    console.log("entering usages")
+
+    const query = `select id, JSON_EXTRACT(config, "$.dimensions[*].variableId") as variables from charts`
+
+    const rows = await db.queryMysql(query)
+
+    const variablesToChartIds = new Map<number, Set<number>>()
+    for (const row of rows) {
+        const variableArray = JSON.parse(row.variables)
+        for (const variable of variableArray) {
+            if (!variablesToChartIds.has(variable)) {
+                variablesToChartIds.set(variable, new Set())
+            }
+            const chartsForVariable = variablesToChartIds.get(variable)
+            chartsForVariable!.add(row.id)
+        }
+    }
+
+    // JSON.stringify supports neither Map nor Set so we have to convert the Map to an object and the Sets to arrays
+    return _.mapValues(Object.fromEntries(variablesToChartIds), (chartIds) => [
+        ...chartIds,
+    ])
```

Comment:
Not sure if it's a good idea API-wise, but we could also just return `variablesToChartIds.entries()` and then simply parse it on the other end using `new Map(...)` without needing `parseInt` and the like?

(would still need to convert the Set, of course)